tools/plugins/com.liferay.ide.sdk.core/src/com/liferay/ide/sdk/core/SDK.java

```diff
@@ -1003,10 +1011,56 @@ public String toXmlString()
         return builder.toString();
     }
 
+    public IStatus reloadProperties() throws CoreException
+    {
+        Map<String,Object> sdkProperties = null;
+        try
+        {
+            sdkProperties = getBuildProperties( true );
+        }
+        catch( Exception e)
+        {
+            throw new CoreException( SDKCorePlugin.createErrorStatus(e.getMessage()));
+        }
+
+        final IStatus sdkValid = validate(sdkProperties);
+
+        if ( sdkValid.isOK() )
+        {
+            BUILD_PROPERTIES = sdkProperties;
+        }
+
+        return sdkValid;
+    }
+
     public IStatus validate()
     {
         MultiStatus status = new MultiStatus( SDKCorePlugin.PLUGIN_ID, IStatus.OK, "", null );
 
+        Map<String,Object> sdkProperties = null;
+        try
+        {
+            sdkProperties = getBuildProperties( false );
+        }
+        catch( Exception e)
+        {
+            status.add( SDKCorePlugin.createErrorStatus( e.getMessage() ) );
+        }
+
+        final IStatus sdkValid = validate(sdkProperties);
+
+        if ( sdkValid.isOK() && BUILD_PROPERTIES == null )
+        {
+            BUILD_PROPERTIES = sdkProperties;
+        }
+
+        return sdkValid;
+    }
+
+    public IStatus validate(final Map<String,Object> sdkProperties)
```

Comment:
I think we can only cache the correct sdk settings not all content get from build.properties should be cached if the lastest content has error in build.properties. But I catch your comments that we don't need to check the build.properties configuration is error or correct.